cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -475,7 +475,6 @@ private String getCursorResponse(MetadataSearchResponse searchResponse) {
 
   private ApplicationId getApplicationId(MetadataEntity metadataEntity) {
     return new ApplicationId(metadataEntity.getValue(MetadataEntity.NAMESPACE),
-                             metadataEntity.getValue(MetadataEntity.APPLICATION),
-                             metadataEntity.getValue(MetadataEntity.VERSION));
+                             metadataEntity.getValue(MetadataEntity.APPLICATION));
```

Comment:
since version is no longer included in the MetadataSearchResponse, we will just construct ApplicationId with the default version.